archive/compression/compression.go

```diff
@@ -139,3 +139,12 @@ func (compression *Compression) Extension() string {
 	}
 	return ""
 }
+
+// WrappedGzipReader checks if r passed-in is a gzip reader wrapped by method DecompressStream
+func WrappedGzipReader(r io.Reader) bool {
```

Comment:
This function should return `Compression`. Maybe a signature like `GetCompression(r. io.Reader) Compression`